### PR TITLE
tests/robustness: Extract validation to separate package

### PR DESCRIPTION
**Pull request by serathius**
cc @ahrtr @ptabor @jmhbnz

### CODE REVIEW

**Comment by serathius** [comment]
Benjamin is out for 2 weeks, any @etcd-io/members wants to review it?

File: tests/robustness/validate/validate.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+	"go.etcd.io/etcd/tests/v3/robustness/model"
+	"go.etcd.io/etcd/tests/v3/robustness/traffic"
+)
+
+// ValidateAndReturnVisualize return visualize as porcupine.linearizationInfo used to generate visualization is private.
+func ValidateAndReturnVisualize(t *testing.T, lg *zap.Logger, cfg Config, reports []traffic.ClientReport) (visualize func(basepath string)) {
+	validateWatch(t, cfg, reports)
+	allOperations := operations(reports)
+	persisted := persistedOperations(reports)
```
Comment (chaochn47) [inline tests/robustness/validate/validate.go:31]: Seems like the `reports` is ranged over both in `operations` and `persistedOperations`. What's your thoughts that those functions are combined together in a single loop just like before?

**Review by chaochn47 — APPROVED** [review]
LGTM with one nit.

**Review by jmhbnz — APPROVED** [review]
LGTM - Happy to have this split out.

One nitpick re consistency of validation functions.

**Comment by serathius** [comment]
> Also need to make fix for goimports.

`make fix` doesn't work on my machine (assume some incompatibilities between my local golangci and Go). Sad that fixing imports requires me to fix golangci. @jmhbnz Could we separate `make fix-goimport`? We had proposed similar effort for contibex.
```
make fix
./scripts/updatebom.sh
generating bill-of-materials.json
% (cd tools/mod && 'go' 'install' 'github.com/coreos/license-bill-of-materials')
% '/home/serathius/bin/license-bill-of-materials' '--override-file' './bill-of-materials.override.json' 'go.etcd.io/etcd/api/v3/...' 'go.etcd.io/etcd/pkg/v3/...' 'go.etcd.io/etcd/client/pkg/v3/...' 'go.etcd.io/etcd/client/v2/...' 'go.etcd.io/etcd/client/v3/...' 'go.etcd.io/etcd/server/v3/...' 'go.etcd.io/etcd/etcdutl/v3/...' 'go.etcd.io/etcd/etcdctl/v3/...' 'go.etcd.io/etcd/tests/v3/...' 'go.etcd.io/etcd/v3/...'
bom refreshed
golangci-lint run --config tools/.golangci.yaml --fix
tools/benchmark/cmd/lease.go:25:2: `v3` redeclared in this block (typecheck)
        "github.com/cheggaaa/pb/v3"
        ^
tools/benchmark/cmd/lease.go:22:2: other declaration of v3 (typecheck)
        v3 "go.etcd.io/etcd/client/v3"
        ^

```

**Comment by jmhbnz** [comment]
> `make fix` doesn't work on my machine (assume some incompatibilities between my local golangci and Go). Sad that fixing imports requires me to fix golangci. @jmhbnz Could we separate `make fix-goimport`? We had proposed similar effort for contibex.

Yeah let me take a look at this.  Agree we don't want any friction with something so basic.

**Pull request merged**